src/main/java/webApp/webController.java

```diff
@@ -55,7 +55,7 @@ public SurveyMessage retrieveSurvey(@RequestParam (name="name") String name) {
 
             for (Question question : surveys.iterator().next().getQuestions()) {
                 if(question.getType().equals("openEnded")) {
-                    questionMessages.add(new QuestionMessage("openEnded", question.getQuestion(), "", ""));
```

Comment:
Instead of checking the Type field of the question, you should check the object type. Use `instanceof` to check if it is OpenEndedQueztion or NumberQuestion.

Then you can remove the Type field from question.